extension/diffheader.js

```diff
@@ -21,6 +21,46 @@ window.diffFileHeader = (() => {
 		};
 	})();
 
+	const maxPxAvailable = () => {
+		// Unfortunately can't cache this value, as it'll change with the browsers zoom level
+		const filenameLeftOffset = $('.diff-toolbar-filename').get(0).getBoundingClientRect().left;
+		const diffStatLeftOffset = $('.diff-toolbar-filename + .right').get(0).getBoundingClientRect().left;
+
+		return diffStatLeftOffset - filenameLeftOffset;
+	};
+
+	const parseFileDetails = filename => {
+		const folderCount = (filename.match(/\//g) || []).length;
+		const [, basename] = (filename.match(/(?:\/)([\w\.-]+)$/) || []);
+		const [, topDir] = (filename.match(/^([\w\.-]+)\//) || []);
+
+		return {
+			folderCount,
+			basename,
+			topDir
+		};
+	};
+
+	const updateFileLabel = val => {
+		const target = $('.diff-toolbar-filename').get(0);
+		target.classList.add('filename-height-check');
+		target.textContent = val;
+
+		const maxPx = maxPxAvailable();
```

Comment:
Could you rename `Px` to `Pixels` everywhere? It could be a bit confusing with the cameCase.